extensions-contrib/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureStorageDruidModule.java

```diff
@@ -91,6 +96,16 @@ public void configure(Binder binder)
     Binders.taskLogsBinder(binder).addBinding(SCHEME).to(AzureTaskLogs.class);
     JsonConfigProvider.bind(binder, "druid.indexer.logs", AzureTaskLogsConfig.class);
     binder.bind(AzureTaskLogs.class).in(LazySingleton.class);
+    binder.install(new FactoryModuleBuilder()
+                       .build(AzureByteSourceFactory.class));
+    binder.install(new FactoryModuleBuilder()
+                       .build(AzureEntityFactory.class));
+    binder.install(new FactoryModuleBuilder()
+                       .build(AzureCloudBlobIteratorFactory.class));
+    binder.install(new FactoryModuleBuilder()
+                       .build(AzureCloudBlobIterableFactory.class));
+    binder.install(new FactoryModuleBuilder()
+                       .build(ListBlobItemDruidFactory.class));
```

Comment:
ModuleTests to ensure the factories are injected and can create their objects?

Look at this PR - #9279 I've added a bunch of module tests with mocks